centrallix-sysdoc/OSDriver_Authoring.md

```diff
@@ -115,497 +170,1032 @@ License:  Copyright (C) 2001-2011 LightSys Technology Services. See LICENSE.txt
 
-    obj->Pathname is a Pathname structure which contains the complete parsed pathname for the object.  This structure is defined in the file include/obj.h, and has a buffer for the pathname as well as an array of pointers to the pathname's components.  The function obj_internal_PathPart() can be used to obtain at will any component or series of components of the pathname.
+- `mask : int`: The permission mask to be given to the object, if it is being created.  Typically, this will only apply to files and directories, so most drivers can ignore it.  The values are the same as the UNIX [octal digit permissions](https://en.wikipedia.org/wiki/Chmod#:~:text=Octal%20digit%20permission) used for the `chmod()` command.
 
-    obj->Pathname->OpenCtl[] contains parameters to the open() operation.  Frequently these params provide additional information on how to open the object.  The use of these parameters is determined by the author of the objectsystem driver.  The parameters are those passed in normal URL fasion (?param=value, etc.). Typically, the only OpenCtl of interest is going to be obj->Pathname->OpenCtl[obj->SubPtr] (see below for SubPtr meaning).
+- `sys_type : pContentType`: Indicates the content type of the node object as determined by the OSML.  The ContentType structure is defined in `include/obj.h`. `sys_type->Name` lists the name of the content type (e.g. `"system/query"` for the query driver).
+<!-- TODO: Greg - What happens if the object doesn't exist yet? Is this NULL? -->
 
-    obj->SubPtr is the number of components in the path that are a part of the node object's path.  For example, in the above path of '/datasources/OMSS_DB', the path would be internally represented as './datasources/ OMSS_DB', and the SubPtr would be 3.
+- `usr_type : char*`: The object type requested by the user.  This is normally used when creating a new object, though some drivers also use it when opening an existing object.  For example, the reporting driver generates HTML report text or plaintext reports if `usr_type` is `"text/html"` or `"text/plain"` (respectively).
 
-    obj->SubCnt reflects the number of components of the path which are under the control of the current driver.  This includes the node object, so SubCnt will always be at least 1.  For example, when opening '/data/file.csv/rows/1', and the driver in question is the CSV driver, SubPtr would be 3 (includes an "invisible" first component), from '/data/file.csv', and SubCnt would be 3, from 'file.csv/rows/1'. The driver will need to SET THE SUBCNT value in its Open function.  SubPtr is already set.
+- `oxt : pObjTrxTree*`: The transaction tree, used when the driver specifies the `OBJDRV_C_TRANS` capability.  More on this field later.  Non-transaction-aware drivers can safely ignore this field.
+  
+  📖 **Note**: Yes, this param *is* a pointer to a pointer.  Essentially, a pointer passed by reference.
```

Comment:
The reason for the pointer passed by reference is so that the driver can initiate the creation of the transaction tree if no transaction is yet in progress.

Reply:
Good to know, updated.